### PR TITLE
Move tab groups logic to store and actions

### PR DESCRIPTION
**Pull request by AnshulMalik**
App.tsx currently does a lot of things that it should not, moving tabs logic out of the component will help in better understanding of the code.

### Summary of Changes

* Change groups to tabGroups
* Move tabs out of App.tsx

### Test Plan

- [x] Open file from left file tree(Single and Double click)
- [x] Split pane
- [x] Close file in either pane

### CODE REVIEW

File: src/actions/AppActions.ts [hunk 1]
```diff
@@ -108,3 +121,76 @@ export function logLn(message: string, kind: "" | "info" | "warn" | "error" = ""
...
+    file
+  } as CloseFileAction);
+}
+
+export interface OpenProjectFilesAction extends AppAction {
+  type: AppActionType.OPEN_PROJECT_FILES;
+  json: ProjectTemplate;
+}
+
+export function openProjectFiles(json: Object) {
```
Comment (yurydelendik) [inline src/actions/AppActions.ts]: make this function async and move:

```
const projectModel = new Project();
const json = await Service.loadProject(project, projectModel);
```
and dispatch `projectModel` and `json.openedFiles`. Notice that ideally we need to create a new project. Similar to load from fiddle (can we use LOAD_PROJECT action for that?)

File: src/actions/AppActions.ts [hunk 2]
```diff
@@ -108,3 +121,76 @@ export function logLn(message: string, kind: "" | "info" | "warn" | "error" = ""
...
+  dispatcher.dispatch({
+    type: AppActionType.OPEN_PROJECT_FILES,
+    json
+  } as OpenProjectFilesAction);
+}
+
+export async function saveProject(fiddle: string) {
+  this.logLn("Saving Project ...");
```
Comment (yurydelendik) [inline src/actions/AppActions.ts]: use logLn

**Comment by yurydelendik** [comment]
Good to go after commit squashing.

**Comment by yurydelendik** [comment]
Thank you for the patch.

**Comment by mbebenita** [comment]
👍

**Pull request merged**